### PR TITLE
[refactor][broker ] PIP-301 Part-2: Add BrokerTimeAverageDataResources

### PR DESCRIPTION
**Pull request by AnonHxy**
### Motivation

See pip: https://github.com/apache/pulsar/pull/21129

### Modifications

Add  `BrokerTimeAverageDataResources`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository
https://github.com/AnonHxy/pulsar/pull/46

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java [hunk 1]
```diff
@@ -69,4 +73,22 @@ private String getBundleDataPath(final String bundle) {
             return BUNDLE_DATA_BASE_PATH + "/" + bundle;
         }
     }
+
+    public static class BrokerTimeAverageDataResources extends BaseResources<TimeAverageBrokerData> {
+        public BrokerTimeAverageDataResources(MetadataStore store, int operationTimeoutSec) {
+            super(store, TimeAverageBrokerData.class, operationTimeoutSec);
+        }
+
+        public CompletableFuture<Void> updateTimeAverageBrokerData(String broker, TimeAverageBrokerData data) {
```
Comment (lifepuzzlefun) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java:82]: maybe we can change args broker to `brokerLookupAddress` or add javadoc to call the api caller how to pass args broker here.

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java [hunk 2]
```diff
@@ -69,4 +73,22 @@ private String getBundleDataPath(final String bundle) {
...
+            return setWithCreateAsync(getTimeAverageBrokerData(broker), __ -> data);
+        }
+
+        public CompletableFuture<Void> deleteTimeAverageBrokerData(String broker) {
+            return deleteAsync(getTimeAverageBrokerData(broker));
+        }
+
+        private String getTimeAverageBrokerData(final String broker) {
```
Comment (lifepuzzlefun) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java:90]: getTimeAverageBrokerData -> getTimeAverageBrokerDataPath much clear.

Comment (AnonHxy) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java:90]: Make sense

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java (hunk 1 above)
Comment (AnonHxy) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/LoadBalanceResources.java:82]: En.. I think that both `brokerLookupAddress` and `broker` is OK.  We can see that other place  also name the arg as `broker`, e.g.:
https://github.com/apache/pulsar/blob/a10564d875f42c32fda0178cfe7ad976d26fbfb1/pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ModularLoadManagerImpl.java#L1179-L1187

**Review by lifepuzzlefun — APPROVED** [review]
LGTM~

**Review by HQebupt — APPROVED** [review]
LGTM

**Pull request merged**